### PR TITLE
Lite 23308 add command to list the locales

### PR DESCRIPTION
**Pull request by jonatrios**

### CODE REVIEW

File: connect/cli/plugins/locale/commands.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+from connect.cli.plugins.locale.constants import LOCALES_TABLE_HEADER
+from connect.cli.plugins.locale.utils import (
+    icon_for_autotranslate,
+    row_format_resource,
+    stats_of_translations,
+)
+from connect.client import ConnectClient, RequestLogger
+
+
+@group(name='locale', short_help='List all locales avaibles.')
```
Comment (qarlosh) [inline connect/cli/plugins/locale/commands.py:21]: typo (availables)

File: connect/cli/plugins/locale/commands.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    name='list',
+    short_help='List locales.',
+)
+@click.option(
+    '--query',
+    '-q',
+    'query',
+    help='RQL query expression.',
+)
+@click.option(
```
Comment (ffaraoneim) [inline connect/cli/plugins/locale/commands.py:36]: I would suggest to use IntRange here https://click.palletsprojects.com/en/8.1.x/api/#click.IntRange

File: connect/cli/plugins/locale/commands.py [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    '--always-continue',
+    '-c',
+    'always_continue',
+    is_flag=True,
+    help='Do not prompt to continue.',
+)
+@pass_config
+def cmd_list_locales(config, query, page_size, always_continue):
+    if page_size < 1:
```
Comment (ffaraoneim) [inline connect/cli/plugins/locale/commands.py:53]: If you will use the https://click.palletsprojects.com/en/8.1.x/api/#click.IntRange this validation will not be needed

**Review by ffaraoneim — CHANGES_REQUESTED** [review]

File: connect/cli/plugins/locale/commands.py (hunk 2 above)
Comment (jonatrios) [reply connect/cli/plugins/locale/commands.py:37]: Thanks for the advice, is updated to use `click.IntRange` for integer validation at `page_size` option now

File: connect/cli/plugins/locale/commands.py (hunk 1 above)
Comment (jonatrios) [reply connect/cli/plugins/locale/commands.py]: Typo fixed.

**Review by marcserrat — APPROVED** [review]

File: connect/cli/plugins/locale/utils.py [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    header = resource_str_list[:1]
+    if paging % page_size == 0:
+        start = 1 if paging == page_size else paging - page_size
+        if paging > page_size:
+            start += 1
+        click.echo(render(''.join(header + resource_str_list[start:])))
+    else:
+        start = count_of_resources - ((paging - 1) % page_size)
+        if resource_str_list[start:] and len(resource_str_list[1:]) == count_of_resources:
+            click.echo(render(''.join(header + resource_str_list[start:])))
```
Comment (qarlosh) [inline connect/cli/plugins/locale/utils.py:30]: For consistency, probably should not use type annotations in just one file of there is not a compelling reason for it.

File: connect/cli/plugins/locale/utils.py [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+from cmr import render
+
+
+def icon_for_autotranslate(resource: Dict[str, Any]) -> str:
+    return '\u2713' if resource['auto_translation'] else '\u2716'
+
+
+def stats_of_translations(resource: Dict[str, Any]) -> Union[str, int]:
+    count = resource["stats"]["translations"]
+    return count if count else '-'
```
Comment (qarlosh) [inline connect/cli/plugins/locale/utils.py:13]: you can use `return count or '-'`, it is shorter and clearer, I think

Comment (jonatrios) [reply connect/cli/plugins/locale/utils.py:13]: Yes you are ritght. Is more readable that way

File: connect/cli/plugins/locale/utils.py (hunk 4 above)
Comment (jonatrios) [reply connect/cli/plugins/locale/utils.py:30]: Ok, I'll take them out in order to mantain consistency.

**Review by ffaraoneim — APPROVED** [review]

**Review by qarlosh — APPROVED** [review]

**Review by d3rky — APPROVED** [review]

**Pull request merged**